index.js

```diff
@@ -1,81 +1,143 @@
+            .on('get', this.getColorTemperature.bind(this))
+            .on('set', this.setColorTemperature.bind(this));
+
+        this.listenLampState().catch(error => this.log.error(error));
     }
 
     async getLamp() {
-        if (this.lampDevice) return this.lampDevice
-        this.log('Connect to device')
+        if (this.lampDevice) return this.lampDevice;
+        this.log('Connect to device');
         try {
-            this.lampDevice = await miio.device({address: this.ip, token: this.token})
+            this.lampDevice = await miio.device({address: this.ip, token: this.token});
         } catch (e) {
-            this.lampDevice = undefined
-            this.log.error('Device not connected', e)
+            this.lampDevice = undefined;
+            this.log.error('Device not connected', e);
         }
-        return this.lampDevice
+        return this.lampDevice;
     }
 
     async listenLampState(){
-        const device = await this.getLamp()
-        device.on('powerChanged', isOn => this.switch.getCharacteristic(Characteristic.On).updateValue(isOn))
+        const device = await this.getLamp();
+        device.on('powerChanged', isOn => this.lamp.getCharacteristic(Characteristic.On).updateValue(isOn));
+	    device.on('colorChanged', color => this.lamp.getCharacteristic(Characteristic.ColorTemperature).updateValue(color));
     }
 
     async getState(callback) {
-        this.log('Get state...')
+        this.log('Get state...');
         try {
-            const device = await this.getLamp()
-            const power = await device.power()
-            callback(null, power)
+            const device = await this.getLamp();
+            const power = await device.power();
+            callback(null, power);
         } catch (e) {
-            this.log.error('Error getting state', e)
-            callback(e)
+            this.log.error('Error getting state', e);
+            callback(e);
         }
     }
 
     async setState(state, callback) {
-        this.log('Set state to', state)
+        this.log('Set state to', state);
+        try {
+            const device = await this.getLamp();
+            await device.power(state);
+            callback(null);
+        } catch (e) {
+            this.log.error('Error setting state', e);
+            callback(e);
+        }
+    }
+
+    async getBrightness(callback) {
+        this.log('Get brightness...');
         try {
-            const device = await this.getLamp()
-            await device.power(state)
-            callback(null)
+            const device = await this.getLamp();
+            const brightness = await device.brightness();
+            callback(null, brightness);
         } catch (e) {
-            this.log.error('Error setting state', e)
-            callback(e)
+            this.log.error('Error getting brightness', e);
+            callback(e);
         }
     }
+	async setBrightness(state, callback) {
```

Comment:
I am not sure if is GitHub, but there is some problems with the code alignement.

Reply:
Probably GitHub. In my IDE it looks ok.